account_invoice_triple_discount/migrations/16.0.2.0.0/post-migrate.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+import openupgrade
+
+
+@openupgrade.logging()
+def compute_discount(env):
+    env["account.move.line"].search(
+        [
+            "|",
+            "|",
+            ("discount1", "!=", 0),
+            ("discount2", "!=", 0),
+            ("discount3", "!=", 0),
+        ]
+    )
+    env["account.move.line"]._compute_discount()
```

Comment:
IMO this will raise an exception for all the lines with a date prior to the fiscal year lock date...